aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,111 @@
-
->[!TIP]
-> A common use case for caching is data-driven navigation menus, which rarely change but are frequently read for display within an application. Caching results that do not vary often but which are requested frequently can greatly improve performance by reducing round trips to out of process data stores and unnecessary computation.
-
-## Configuring In Memory Caching
-
-To use an in memory cache in your ASP.NET application, add the following dependencies to your *project.json* file:
-
-[!code-json[Main](memory/sample/src/CachingSample/project.json?range=7-13&highlight=4)]
-
-Caching in ASP.NET Core is a *service* that should be referenced from your application by [Dependency Injection](../../fundamentals/dependency-injection.md). To register the caching service and make it available within your app, add the following line to your `ConfigureServices` method in `Startup`:
+## Caching basics
 
-[!code-csharp[Main](memory/sample/src/CachingSample/Startup.cs?range=12-15&highlight=3)]
+Caching can significantly improve the performance and scalability of an app by reducing round trips to the data store. Caching works best with data that changes infrequently. Caching makes a copy of data that can be returned much faster than from the original source. You should write and test your app to never depend on cached data.
 
-You utilize caching in your app by requesting an instance of `IMemoryCache` in your controller or middleware constructor. In the sample for this article, we are using a simple middleware component to handle requests by returning customized greeting. The constructor is shown here:
+ASP.NET Core supports several different caches. The simplest cache is based on the [IMemoryCache](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache), which represents a cache stored in the memory of the web application. Apps that utilize scale out in the cloud (run on multiple servers, that is a server farm), should ensure that sessions are sticky when using the in-memory cache. Sticky sessions ensure that subsequent requests from a client all go to the same server. For example, Azure Web apps use [Application Request Routing](http://www.iis.net/learn/extensions/planning-for-arr) (ARR) to route all subsequent requests to the same server.
 
-[!code-csharp[Main](memory/sample/src/CachingSample/Middleware/GreetingMiddleware.cs?range=19-28&highlight=2,7)] 
+If an app is hosted by multiple servers in a web farm or cloud hosting environment, and you cannot guarantee sticky sessions (that is subsequent requests to the same server), you will not be able to use the in-memory cache. Non-sticky sessions require a [distributed cache](distributed.md) to avoid cache consistency problems. 
```

Comment:
Hmm the language is a little strict here. Technically you can still use in memory cache even when sticky sessions are not guaranteed if you do not care about consistency. I would recommend using should instead: "... you should not use the in-memory cache. Non-sticky sessions should use a [distributed cache](distributed.md) to avoid cache consistency problems."